### PR TITLE
rootless: do not set pids limits with cgroupfs

### PR DESCRIPTION
**Pull request by giuseppe**
and enable events tests.

Signed-off-by: Giuseppe Scrivano <gscrivan@redhat.com>

### CODE REVIEW

**Comment by rhatdan** [comment]
LGTM

**Comment by mheon** [comment]
LGTM

**Comment by mheon** [comment]
Haven't seen this test failure before, might be legitimate

File: cmd/podman/common/specgen.go [hunk 1]
```diff
@@ -128,18 +130,21 @@ func getIOLimits(s *specgen.SpecGenerator, c *ContainerCLIOpts, args []string) (
...
+		pids.Limit = containerConfig.PidsLimit()
+		return pids, nil
+	}
 	if c.PIDsLimit > 0 {
 		pids.Limit = c.PIDsLimit
-		hasLimits = true
-	}
-	if !hasLimits {
-		return nil, nil
+		return pids, nil
```
Comment (TomSweeneyRedHat) [inline cmd/podman/common/specgen.go:145]: if `c.PidsLimit == 0` is falling through and returning `nil, nil` the right thing?

Comment (giuseppe) [reply cmd/podman/common/specgen.go:145]: yes, `PidsLimit==0` means no limits

File: cmd/podman/common/specgen.go [hunk 2]
```diff
@@ -128,18 +130,21 @@ func getIOLimits(s *specgen.SpecGenerator, c *ContainerCLIOpts, args []string) (
 
 func getPidsLimits(s *specgen.SpecGenerator, c *ContainerCLIOpts, args []string) (*specs.LinuxPids, error) {
```
Comment (rhatdan) [inline cmd/podman/common/specgen.go:131]: It does not look like this function returns an error, should we drop the error?

Comment (giuseppe) [reply cmd/podman/common/specgen.go]: sure, pushed a new version.

**Comment by giuseppe** [comment]
tests pass now

File: test/e2e/run_networking_test.go [hunk 3]
```diff
@@ -202,7 +203,7 @@ var _ = Describe("Podman run networking", func() {
 		Expect(inspect.ExitCode()).To(Equal(0))
 
 		containerConfig := inspect.InspectContainerToJSON()
-		Expect(containerConfig[0].NetworkSettings.Ports[0].HostPort).ToNot(Equal("80"))
+		Expect(containerConfig[0].NetworkSettings.Ports[0].HostPort).ToNot(Equal(80))
```
Comment (jwhonce) [inline test/e2e/run_networking_test.go:206]: Is this an expected change?  Seems like it could break clients.

**Pull request merged**